src/frontend/components/TimedTextCreationForm/TimedTextCreationForm.tsx

```diff
@@ -0,0 +1,163 @@
+import * as React from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import { Redirect } from 'react-router-dom';
+import Select from 'react-select';
+import { ActionMeta, ValueType } from 'react-select/lib/types';
+import styled from 'styled-components';
+
+import { createTimedTextTrack } from '../../data/sideEffects/createTimedTextTrack/createTimedTextTrack';
+import { getTimedTextTrackLanguageChoices } from '../../data/sideEffects/getTimedTextTrackLanguageChoices/getTimedTextTrackLanguageChoices';
+import { modelName } from '../../types/models';
+import { TimedText, timedTextMode } from '../../types/tracks';
+import { colors } from '../../utils/theme/theme';
+import { Maybe, Nullable } from '../../utils/types';
+import { Button } from '../Button/Button';
+import { ERROR_COMPONENT_ROUTE } from '../ErrorComponent/route';
+import { UPLOAD_FORM_ROUTE } from '../UploadForm/route';
+
+const messages = defineMessages({
+  addTrackBtn: {
+    defaultMessage: 'Upload the file',
+    description:
+      'Text for the button in the dashboard to go upload a new timed text file.',
+    id: 'components.TimedTextCreationForm.addTrackBtn',
+  },
+  addTrackLabel: {
+    defaultMessage: 'Add a language',
+    description:
+      'Label for the form that lets the instructor add a new timed text track.',
+    id: 'components.TimedTextCreationForm.addTrackLabel',
+  },
+  error: {
+    defaultMessage: 'There was an error during track creation.',
+    description:
+      'Generic error message when we fail to create a subtitle/transcript/caption track.',
+    id: 'components.TimedTextCreationForm.error',
+  },
+});
+
+const TimedTextCreationFormStyled = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  margin-top: 1rem;
+`;
+
+const ErrorMessage = styled.div`
+  color: ${colors.danger.main};
+`;
+
+interface SelectOption {
+  label: string;
+  value: string;
+}
+
+/** Props shape for the TimedTextCreationForm component. */
+export interface TimedTextCreationFormProps {
+  createTimedTextTrack: (timedtexttrack: TimedText) => void;
+  jwt: Nullable<string>;
+  mode: timedTextMode;
+}
+
+/** State shape for the TimedTextCreationForm component. */
+interface TimedTextCreationFormState {
+  error?: 'creation' | 'schema';
+  languageChoices: SelectOption[];
```

Comment:
I think it is the responsibility of this component to retrieve this list.

Sure, we can optimize it by moving the call up in the component tree, but in my opinion the better way to do this is to add caching for the call itself.

Since the side effects are to be replaced with a cleaner structure through `saga` or `thunks` I chose not to implement that in this PR.